library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -46,7 +46,7 @@
    * Invokes onCreateHeaderViewHolder, onCreateItemViewHolder or onCreateFooterViewHolder methods
    * based on the view type param.
    */
-  @Override public VH onCreateViewHolder(ViewGroup parent, int viewType) {
+  @Override public final VH onCreateViewHolder(ViewGroup parent, int viewType) {
```

Comment:
You said:
"These methods just check if the view type or the position is related to a header. Making these method finals the library clients can override onCreateItemViewHolder and onBindItemViewHolder to implement custom bindings."
here https://github.com/Karumi/HeaderRecyclerView/issues/3#issuecomment-106264010

So did I misunderstand your meaning?

If the user want to use different view types but the same data structure, they can simply dectect what item type it is in onCreateItemViewHolder, and create the needed view. But if they want to create something like header2 or footer2, mark onCreateViewHolder as final will compel them to put header2 and footer2 to items, even though they don't have any common point with items. 

So declare onCreateViewHolder as final will restrict users to add additional special view types(not items), right?

Em, therefore, #3 should be closed, and I will also make onCreateViewHolder not final.
(∩_∩)